include/mxnet/imperative.h

```diff
@@ -160,8 +160,8 @@ class Imperative {
 
  private:
   friend class NDArray;
-  /*! \brief make constructor protected. */
-  Imperative() {
+  Imperative() :
+    backward_graph_dump_enabled_(dmlc::GetEnv("MXNET_BACKWARD_GRAPH_DUMP", false)) {
```

Comment:
Why do we need a class member for a debugging utility?

Reply:
to access the environment only once.

Reply:
If this is only used for debugging, I am not sure if adding a field to a production class is a good idea. Can we just pull the environment variable instead? It's not impacting production performance and not going to be used in multithreaded environment.